extensions/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelModelJAXBContextFactoryBuildItem.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.core.deployment;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+import io.quarkus.runtime.RuntimeValue;
+import org.apache.camel.spi.ModelJAXBContextFactory;
+
+/**
+ * Holds the {@link ModelJAXBContextFactory} {@link RuntimeValue}.
```

Comment:
General question: The `value` field is not `RuntimeValue<ModelJAXBContextFactory>` (so the JavaDoc should eventually be adapted to match the reality) but the question is whether it should be the case. IIRC, the general rule in Quarkus is to instantiate the runtime classes only within a recorder and thus wrap them in a `RuntimeValue` in BuildItems. I recently discussed this with @lburgazzoli and we came to a conclusion that instantiation in a BuildProcessor may work for types that do not have any complex dependencies (that's probably the current case) but still I was not able to find an example of such a practice in Quarkus. So the question is what are the criteria for you to instantiate a runtime class in a BuildStep vs. in a Recorder?

Reply:
So I initially tried with a `RuntimeValue<>` but it did not work.  I think the use of the `@Overridable` does not play well with `@Record`, so I had to remove the use of a `Recorder` and thus the use of the `RuntimeValue`.

Reply:
Can you report the issue up to quarkus when you have time ?

Reply:
Done at https://github.com/quarkusio/quarkus/issues/4479